raiden/transfer/channel.py

```diff
@@ -122,10 +122,19 @@ def is_lock_locked(
 
 
 def is_lock_expired(
+        end_state: NettingChannelEndState,
         locked_lock: typing.Optional[LockedTransferSignedState],
         secrethash: typing.SecretHash,
         block_number: typing.BlockNumber,
 ):
+    """ Determine whether a lock is expired based on certain conditions. """
+
+    # If secret is registered on-chain, the lock never expires.
+    secret_registered_on_chain = secrethash in end_state.secrethashes_to_onchain_unlockedlocks
```

Comment:
note: this assumes the lock is moved to  `secrethashes_to_onchain_unlockedlocks` only if the secret was registered before the lock expired, this needs to be checked

Reply:
One idea is to check if the lock expired on `ReceiveSecretReveal` or `ContractReceiveSecretReveal` (in target.py) they're handled in the same function. If lock has expired, we could just delete the lock instead of moving it to `secrethashes_to_onchain_unlockedlocks`. But in this case, if the secret was accepted and registered by the contract, this solution would be wrong.
The alternative is to get the block number at which the secret was registered on chain and then check against lock expiration block number in `is_lock_expired` above... 
which one are you suggesting?